src/main/java/core/basesyntax/model/lib/Injector.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.model.lib;
+
+import core.basesyntax.model.dao.BetDao;
+import core.basesyntax.model.dao.UserDao;
+import core.basesyntax.model.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException,
+            InstantiationException, RuntimeException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+
+        Field[] declaredFields = clazz.getDeclaredFields();
+
+        for (Field field : declaredFields) {
+            field.setAccessible(true);
+            if (field.getAnnotation(Inject.class) != null) {
+                if (field.getType().equals(BetDao.class)
+                        && BetDao.class.isAnnotationPresent(Dao.class)) {
+                    field.set(instance, Factory.getBetDao());
+                } else if (field.getType().equals(UserDao.class)
+                        && UserDao.class.isAnnotationPresent(Dao.class)) {
+                    field.set(instance, Factory.getUserDao());
+                } else {
+                    throw new NoDaoAnnotationException(field.getName()
+                            + " doesn't contain @Dao annotation");
+                }
```

Comment:
looks good
I would change the name of your custom Exception

```suggestion
                if (field.getType().equals(BetDao.class)
                        && Factory.getBetDao().getClass().isAnnotationPresent(Dao.class)) {
                    field.set(instance, Factory.getBetDao());
                } else if (field.getType().equals(UserDao.class)
                        && Factory.getUserDao().getClass().isAnnotationPresent(Dao.class)) {
                    field.set(instance, Factory.getUserDao());
                } else {
                    throw new NoImplementationException(field.getName()
                            + " doesn't have an implementation that can be injected");
                }
```

Reply:
done